pynsot/models.py

```diff
@@ -7,98 +7,550 @@
+            if self.exists() and self.rtype != 'interfaces':
+                self.logger.debug('[%s] Patching', self.identifier)
+                to_ensure['id'] = self.existing_resource()['id']
+                self.rclient.patch([to_ensure])
+                self.logger.info('[%s] has been patched!', self.identifier)
+                self.clear_cache()
+                return True
+
+            else:  # POST for initially creating
+                self.logger.debug('[%s] Creating', self.identifier)
+                self.rclient.post([to_ensure])
+                self.logger.info('[%s] has been created!', self.identifier)
+                return True
+        except Exception as e:
+            self.log_error(e)
+            self.clear_cache()
+            return False
+
+    def purge(self):
+        '''Delete resource from NSoT
+
+        Deletes the current resource upstream by site and identifier
+        '''
+        # Clears state before starting
+        self.clear_cache()
+        try:
+            if self.exists():
+                self.logger.debug('[%s] Deleting', self.identifier)
+                id_ = self.existing_resource()['id']
+                self.rclient(id_).delete()
+                self.logger.info('[%s] has been deleted!', self.identifier)
+                self.clear_cache()
+                return True
+
+            else:
+                # We should return True to show the state is already how we
+                # want it to be. False should be for unable to ensure resource
+                # is gone
+                return True
+        except Exception as e:
+            self.log_error(e)
+            self.clear_cache()
+            return False
+
+
+class Network(Resource):
+    '''Network Resource
+
+    Subclass of Resource.
+
+    >>> n = Network(CLIENT, 1, network='8.8.8.0/32')
+    >>> n.exists()
+    False
+    >>> n.ensure()
+    True
+    >>> n.exists()
+    True
+    >>> n.existing_resource()
+    {u'attributes': {},
+     u'id': 31,
+     u'ip_version': u'4',
+     u'is_ip': True,
+     u'network_address': u'8.8.8.0',
+     u'parent_id': 1,
+     u'prefix_length': 32,
+     u'site_id': 1,
+     u'state': u'assigned'}
+
+    >>> n.purge()
+    True
+    >>> n.exists()
+    False
+
+    Attributes:
+        self.net (str): Network
+        self.mask (str): Netmask
+    Options:
+        network (str): (kwarg) network in the form of "IP/NM"
+    '''
+
+    def postinit(self, cidr=None):
+        if not any([cidr, self.raw]):
+            raise TypeError('Networks require a cidr')
+        net = IPNetwork(cidr)
+
+        ver = net.version
+        self.network_address = str(net.network)
+        self.prefix_length = int(net.prefixlen)
+        self.is_host = ver == 4 and net.prefixlen == 32 or net.prefixlen == 128
+        self.init_payload()
+
+    @property
+    def identifier(self):
+        return '%s/%d' % (self['network_address'], self['prefix_length'])
+
+    @property
+    def rclient(self):
```

Comment:
Ah... It's resource-specific! Can we please rename it to `resource_client`, or in keeping w/ how it's done elsewhere in `pynsot`: `self.resource`?